src/collection-proxy.js

```diff
@@ -416,6 +416,17 @@ class CollectionProxy {
     return collection
   }
 
+  /**
+  * Returns reversed version of the original collection
+  *
+  * @returns {Array}
+  */
+  reverse () {
+    this._enqueue('reverse')

```

Comment:
My gut feeling is that you should clone the collection to ensure that the reverse operation doesn't affect the original array.

```suggestion
    this.clone()._enqueue('reverse')
```